syntax = "proto3";

option optimize_for = LITE_RUNTIME;

package zcash;

message GetAddressUtxosRequest {
  repeated string addresses = 1;
  uint64 startHeight = 2;
  uint32 maxEntries = 3;  // zero means unlimited
}

message ZCashUtxo {
  string address = 6;
  bytes txid = 1;
  int32 index = 2;
  bytes script = 3;
  int64 valueZat = 4;
  uint64 height = 5;
}

message GetAddressUtxosResponse {
  repeated ZCashUtxo addressUtxos = 1;
}
